internal/config/config.go

```diff
@@ -0,0 +1,116 @@
+package config
+
+import (
+	"fmt"
+	"github.com/kelseyhightower/envconfig"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+)
+
+const (
+	TESTNET_MAGIC = 1097911063
+	MAINNET_MAGIC = 764824073
+)
+
+type Config struct {
+	Logging LoggingConfig `yaml:"logging"`
+	Api     ApiConfig     `yaml:"api"`
+	Node    NodeConfig    `yaml:"node"`
+}
+
+type LoggingConfig struct {
+	Level string `yaml:"level" envconfig:"LOGGING_LEVEL"`
+}
+
+type ApiConfig struct {
+	ListenAddress string `yaml:"address" envconfig:"API_LISTEN_ADDRESS"`
+	ListenPort    uint   `yaml:"port" envconfig:"API_LISTEN_PORT"`
+}
+
+type NodeConfig struct {
+	Network      string `yaml:"network" envconfig:"NODE_NETWORK"`
+	NetworkMagic uint32 `yaml:"networkMagic" envconfig:"NODE_NETWORK_MAGIC"`
+	Address      string `yaml:"address" envconfig:"NODE_ADDRESS"`
+	Port         uint   `yaml:"port" envconfig:"NODE_PORT"`
+	SocketPath   string `yaml:"socketPath" envconfig:"NODE_SOCKET_PATH"`
```

Comment:
These should use the "standard" Cardano variables, IMO... `CARDANO_NODE_SOCKET_PATH` is used in the upstream scripts. If we're going to have environment variables, they might as well match what the entrypoint scripts do in the cardano-submit-api image.

Reply:
Updated